src/librbd/io/ObjectRequest.cc

```diff
@@ -246,6 +253,22 @@ void ObjectReadRequest<I>::handle_read_object(int r) {
     return;
   }
 
+  // merge ExtentResults to a single sparse bufferlist
+  int pos = 0;
+  for (auto& [read_data, extent_map] : m_extent_results) {
```

Comment:
Nit: at some point it might be a good idea to add an assertion that the extents are in ascending order to avoid the possibility of weird data corruption.

Reply:
Are you referring to the input extents given to ObjectReadRequest? Or the extents returned by rados sparse_read? Or both?

Reply:
Honestly -- both. It would be a bug in librbd if the input extents are out-of-order and it would be a bug in the OSD if the sparse read extents are out-of-order.

Reply:
I will add an assert on the extent map going out back from ObjectReadRequest.
As it is merged from the result of the extent maps coming back from librados, it should cover all.